bridge/engine/EngineFunction.h

```diff
@@ -0,0 +1,254 @@
+#include <cassert>
+#include <tuple>
+
+namespace bridge
+{
+
+class EngineFunction;
+
+namespace detail
+{
+
+constexpr size_t calculateStorageSize()
+{
+    constexpr size_t minSize = 128;
+    // EngineFunction will have EngineFunctionStorage + a pointer
+    // Most likely std::max_align_t will return 16. Then it would create
+    // padding of 8 bytes for a 128 size storage. Here we will ensure that we
+    // take advantage of all allocated memory without wasting memory with paddings
+    // when we have a contiguous collection of EngineFunction
+    constexpr size_t alignedSpace =
+        (minSize + sizeof(void*) + (alignof(std::max_align_t) - 1)) & ~(alignof(std::max_align_t) - 1);
+
+    return alignedSpace - sizeof(void*);
+}
+
+using EngineFunctionStorage = std::aligned_storage_t<calculateStorageSize(), alignof(std::max_align_t)>;
+
+class Invokable
+{
+public:
+    virtual ~Invokable() = default;
+
+private:
+    friend class ::bridge::EngineFunction;
+    virtual void invoke() const = 0;
+    virtual Invokable* moveTo(EngineFunctionStorage& newStorage) = 0;
+    virtual Invokable* copyTo(EngineFunctionStorage& newStorage) const = 0;
+};
+
+template <class Func, class... Args>
+class EngineFunctionBinder final : public Invokable
```

Comment:
Should we not rename this and move it outside bridge? It is generically reusable and the EngineFunction could be defined as a type of the GenericFunction (will ponder a bit more on the name) and specification of the storage size.

Reply:
utils::Function maybe would work.